net-p2p/retroshare/patches/retroshare-0.6.4.patchset

```diff
@@ -0,0 +1,192 @@
+From 55856cf2b700b19d4fdae1665520ff87610a7d30 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
+Date: Mon, 9 Apr 2018 23:50:00 +0200
+Subject: Haiku supporting patches
+
+
+diff --git a/libretroshare/src/libretroshare.pro b/libretroshare/src/libretroshare.pro
+index c2604fa..de27c8c 100644
+--- a/libretroshare/src/libretroshare.pro
++++ b/libretroshare/src/libretroshare.pro
+@@ -365,7 +365,7 @@ openbsd-* {
+ 
+ haiku-* {
+ 
+-	QMAKE_CXXFLAGS *= -Dfseeko64=fseeko -Dftello64=ftello -Dstat64=stat -Dstatvfs64=statvfs -Dfopen64=fopen
++	QMAKE_CXXFLAGS *= -Dfseeko64=fseeko -Dftello64=ftello -Dstat64=stat -Dstatvfs64=statvfs -Dfopen64=fopen -D_BSD_SOURCE
+ 	OPENPGPSDK_DIR = ../../openpgpsdk/src
+ 	INCLUDEPATH *= $${OPENPGPSDK_DIR} ../openpgpsdk
+ 	DEFINES *= NO_SQLCIPHER
+diff --git a/libretroshare/src/rsserver/rsaccounts.cc b/libretroshare/src/rsserver/rsaccounts.cc
+index 6136a2a..3ce9ea0 100644
+--- a/libretroshare/src/rsserver/rsaccounts.cc
++++ b/libretroshare/src/rsserver/rsaccounts.cc
+@@ -348,7 +348,11 @@ bool RsAccountsDetail::defaultBaseDirectory()
+ 	}
+ 
+ 	basedir = h;
+-	basedir += "/.retroshare";
++	#ifndef __HAIKU__
++		basedir += "/.retroshare";
++	#else
++		basedir += "/config/settings/retroshare";
++	#endif
+ 
+ #else
+ 	if (RsInit::isPortable())
+@@ -811,6 +815,8 @@ static bool checkAccount(const std::string &accountdir, AccountDetails &account,
+ 	dataDirectory = ".";
+ #elif defined(ANDROID)
+ 	dataDirectory = defaultBaseDirectory()+"/usr/share/retroshare";
++#elif defined(__HAIKU__)
++	dataDirectory = ".";
+ #elif defined(DATA_DIR)
+ 	// cppcheck-suppress ConfigurationNotChecked
+ 	dataDirectory = DATA_DIR;
+diff --git a/libretroshare/src/rsserver/rsinit.cc b/libretroshare/src/rsserver/rsinit.cc
+index 0d93002..1becc1b 100644
+--- a/libretroshare/src/rsserver/rsinit.cc
++++ b/libretroshare/src/rsserver/rsinit.cc
+@@ -1228,6 +1228,9 @@ int RsServer::StartupRetroShare()
+ #ifdef __APPLE__
+ 	plugins_directories.push_back(rsAccounts->PathDataDirectory()) ;
+ #endif
++#ifdef __HAIKU__
++	plugins_directories.push_back("/boot/system/apps/RetroShare/plugins") ;
```

Comment:
It would be nice if [`find_path()`][0] is used instead. It would let us handle the case where the package is installed in user prefix.